src/machine/machine_atsamd21.go

```diff
@@ -2209,3 +2209,69 @@ func ResetProcessor() {
 
 	arm.SystemReset()
 }
+
+// DAC on the SAMD21.
+type DAC struct {
+}
+
+var (
+	DAC0 = DAC{}
+)
+
+// DACConfig placeholder for future expansion.
+type DACConfig struct {
+}
+
+// Configure the DAC.
+// output pin must already be configured.
+func (dac DAC) Configure(config DACConfig) {
+	// Turn on clock for DAC
+	sam.PM.APBCMASK.SetBits(sam.PM_APBCMASK_DAC_)
+
+	// Use Generic Clock Generator 0 as source for DAC.
+	sam.GCLK.CLKCTRL.Set((sam.GCLK_CLKCTRL_ID_DAC << sam.GCLK_CLKCTRL_ID_Pos) |
+		(sam.GCLK_CLKCTRL_GEN_GCLK0 << sam.GCLK_CLKCTRL_GEN_Pos) |
+		sam.GCLK_CLKCTRL_CLKEN)
+	waitForSync()
+
+	// reset DAC
+	sam.DAC.CTRLA.Set(sam.DAC_CTRLA_SWRST)
+	syncDAC()
+
+	// wait for reset complete
+	for sam.DAC.CTRLA.HasBits(sam.DAC_CTRLA_SWRST) {
+	}
+
+	// enable
+	sam.DAC.CTRLB.Set(sam.DAC_CTRLB_EOEN | sam.DAC_CTRLB_REFSEL_AVCC)
+	sam.DAC.CTRLA.Set(sam.DAC_CTRLA_ENABLE)
+}
+
+// WriteData writes a single value to the DAC.
+func (dac DAC) WriteData(value uint16) error {
+	syncDAC()
+	sam.DAC.DATA.Set(value & 0x3FF)
+	syncDAC()
+	sam.DAC.CTRLA.SetBits(sam.DAC_CTRLA_ENABLE)
+	syncDAC()
+	return nil
+}
+
+// Write is intended to satisfy the Writer interface.
+// This current implementation is probably not very useful.
+func (dac DAC) Write(p []uint16) (n int, err error) {
```

Comment:
This does not satisfy the `Writer` interface, in fact it breaks this support because the name is the same while the signature is different. Maybe it's better to leave it out until a good use case can be found.

Reply:
Removed.